undate/undate.py

```diff
@@ -92,16 +97,25 @@ class UndateInterval:
     # date range between two uncertain dates
 
     def __init__(
-        self, earliest: Union[Undate, None] = None, latest: Union[Undate, None] = None
+        self,
+        earliest: Union[Undate, None] = None,
+        latest: Union[Undate, None] = None,
+        label: Union[str, None] = None,
     ):
         # for now, assume takes two undate objects
         self.earliest = earliest
         self.latest = latest
+        self.label = label
 
     def __str__(self) -> str:
         # using EDTF syntax for open ranges
         return "%s/%s" % (self.earliest or "..", self.latest or "")
```

Comment:
What do you think about including the label in the string representation also? I was thinking if it was set it would take precedence, e.g. "time period (dates)" — but I don't know if that is potentially problematic for parsing / exporting dates.  I guess some of our target export formats may support labels and others may not...

Reply:
Ya, I'm not sure. Currently `str` returns a defined format (EDTF), right? Adding labels would mean you couldn't use the string representation when you want that format, but you would need an extra formatter. I'm thinking of cases when you create a UndateInterval and then use another library to do something with the string representation. Instead of just turning it into a string you would now have to use an extra formatter. Does that make sense?

Reply:
Yes, after thinking about it more, I agree with you — label shouldn't be included in the str method unless you're using a format that supports it.